src/stp/new_plays/GenericPass.cpp

```diff
@@ -18,20 +20,26 @@
 
 namespace rtt::ai::stp::play {
 
+void GenericPass::onInitialize() noexcept {
+    passerShot = false;
+    passingPosition = calculatePassLocation(Grid(0, -2.5, 2, 2.5, 5, 5)).first;
+}
+
 GenericPass::GenericPass() : Play() {
     startPlayInvariants.clear();
     startPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
-    startPlayInvariants.emplace_back(std::make_unique<invariant::BallCloseToUsInvariant>());
+    //startPlayInvariants.emplace_back(std::make_unique<invariant::BallCloseToUsInvariant>());
     startPlayInvariants.emplace_back(std::make_unique<invariant::BallOnOurSideInvariant>());
+    startPlayInvariants.emplace_back(std::make_unique<invariant::BallClosestToUsInvariant>());
 
     keepPlayInvariants.clear();
     keepPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
     keepPlayInvariants.emplace_back(std::make_unique<invariant::FreedomOfRobotsInvariant>());
 
     roles = std::array<std::unique_ptr<Role>, rtt::ai::Constants::ROBOT_COUNT()>{std::make_unique<role::Keeper>(role::Keeper("keeper")),
                                                                                  std::make_unique<role::Passer>(role::Passer("passer")),
-                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver")),
-                                                                                 std::make_unique<role::Halt>(role::Halt("halt_0")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_left")),
+                                                                                 std::make_unique<role::PassReceiver>(role::PassReceiver("receiver_right")),
                                                                                  std::make_unique<role::Halt>(role::Halt("halt_1")),
                                                                                  std::make_unique<role::Halt>(role::Halt("halt_2")),
                                                                                  std::make_unique<role::Halt>(role::Halt("halt_3")),
```

Comment:
All of these you can just make_unique<Keeper>("keeper"), you don't have to explicitly construct one.

Reply:
How though, the compiler doesn't know what type to deduce from a string "keeper"